recipes/spdlog/all/conanfile.py

```diff
@@ -45,46 +40,45 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
+        if self.options.shared or self.options.header_only:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
         if self.options.header_only:
             del self.options.shared
-            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        # TODO: Remove in Conan 2.0 - 1.x self.requires does not support transitive_headers
-        requires = lambda ref: self.requires(ref, transitive_headers=True) if Version(conan_version) >= "2.0.0-beta" else self.requires(ref)
         if Version(self.version) >= "1.10.0":
-            requires("fmt/8.1.1")
+            self.requires("fmt/8.1.1", transitive_headers=True)
         elif Version(self.version) >= "1.9.0":
-            requires("fmt/8.0.1")
+            self.requires("fmt/8.0.1", transitive_headers=True)
         elif Version(self.version) >= "1.7.0":
-            requires("fmt/7.1.3")
+            self.requires("fmt/7.1.3", transitive_headers=True)
         elif Version(self.version) >= "1.5.0":
-            requires("fmt/6.2.1")
+            self.requires("fmt/6.2.1", transitive_headers=True)
         else:
-            requires("fmt/6.0.0")
+            self.requires("fmt/6.0.0", transitive_headers=True)
```

Comment:
@uilianries you have added these `transitive_headers`, but therefore why not `transitive_libs`?

Reply:
libs should be based on shared option of this recipe... ie `transitive_libs = not shared` so the default Conan logic should be enough AFAIK. headers are always false because good coding practices should be encouraged 😉

Reply:
when headers are transitive, it means that libs are transitive as well usually.

Also you are talking about some conan v2 logic where transitive is also based on static/shared option. I hope it doesn't work like that since shared=True doesn't mean that all libs of a recipe are shared.

Reply:
@SpaceIm because it was not needed I built, I talked to Conan devs when I needed to use transitive headers, but didn't mention about the need of transitive libs.